File: algorithms/other/01-fibonacci/solution.ts
/**
* Fibonacci to generate array with length of num.
  npx ts-node solution.ts
*/

function fibonacciTS(num: number) {
  const array: Array<number> = [];
  let n1: number = 0;
  let n2: number = 1;
  let n3: number;
  for (let i = 0; i < num; i++) {
    array.push(n1);
    n3 = n1 + n2;
    n1 = n2;
    n2 = n3;

  }
  return array
}

console.log(fibonacciTS(7))

File: algorithms/other/01-fibonacci/solution.c
/**
 * https://www.tutorialspoint.com/format-specifiers-in-c
 * ctrl+shift+c
 * cd algorithms/01-fibonacci 
 * solution
*/

#include <stdio.h>
int main() {

  int n;
  int n1 = 0;
  int n2 = 1;

  int n3 = n1 + n2;

  printf("Enter the number of terms: ");
  // Signed integer %d
  
  scanf("%d", &n);

  printf("Fibonacci Series: %d, %d, ", n1, n2);

  for (int i = 3; i <= n; ++i) {
    printf("%d, ", n3);
    n1 = n2;
    n2 = n3;
    n3 = n1 + n2;
  }
}
